package/gluon-mesh-vpn-core/src/respondd.c

```diff
@@ -0,0 +1,94 @@
+  All rights reserved.
+  Redistribution and use in source and binary forms, with or without
+  modification, are permitted provided that the following conditions are met:
+    1. Redistributions of source code must retain the above copyright notice,
+       this list of conditions and the following disclaimer.
+    2. Redistributions in binary form must reproduce the above copyright notice,
+       this list of conditions and the following disclaimer in the documentation
+       and/or other materials provided with the distribution.
+  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+  FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+  DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+  SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+  OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+
+#include <respondd.h>
+
+#include <json-c/json.h>
+#include <uci.h>
+
+#include <stdbool.h>
+#include <string.h>
+#include <unistd.h>
+
+
+static struct json_object * get_simple_tc(void) {
+	bool enabled = false;
+	int egress = 0;
+	int ingress = 0;
+
+	struct json_object *ret = json_object_new_object();
+
+	struct uci_context *ctx = uci_alloc_context();
+	if (!ctx)
+		goto disabled_nofree;
+	ctx->flags &= ~UCI_FLAG_STRICT;
+
+	struct uci_package *p;
+	if (uci_load(ctx, "simple-tc", &p))
+		goto disabled;
+
+	struct uci_section *s = uci_lookup_section(ctx, p, "mesh_vpn");
+	if (!s)
+		goto disabled;
+
+	const char *enabled_str = uci_lookup_option_string(ctx, s, "enabled");
+	if (!enabled_str || !strcmp(enabled_str, "1"))
+		enabled = true;
+
+	const char *egress_str = uci_lookup_option_string(ctx, s, "limit_egress");
+	egress = atoi(egress_str);
+
+	const char *ingress_str = uci_lookup_option_string(ctx, s, "limit_ingress");
+	ingress = atoi(ingress_str);
+
+disabled:
+	uci_free_context(ctx);
+
+disabled_nofree:
+	json_object_object_add(ret, "enabled", json_object_new_boolean(enabled));
+	if (enabled) {
+		json_object_object_add(ret, "limit_ingress", json_object_new_int(ingress));
+		json_object_object_add(ret, "limit_egress", json_object_new_int(egress));
+	}
+	return ret;
+}
+
+
+static struct json_object * respondd_provider_nodeinfo(void) {
+	struct json_object *ret = json_object_new_object();
+	struct json_object *network = json_object_new_object();
+	struct json_object *mesh_vpn = json_object_new_object();
+
+	json_object_object_add(mesh_vpn, "simple_tc", get_simple_tc());
+	json_object_object_add(network, "mesh_vpn", mesh_vpn);
```

Comment:
`mesh_vpn`was intended to match `statistics.mesh_vpn` not the interface `mesh-vpn`, that (as unrelated sidenote) might be renamed to `vpn` in #2168.

I think it's unlikely we are going to use simple-tc on other interfaces than the uplink, don't you think?

Reply:
> Or maybe call it "simple-tc"?

I think using dashes instead of underscores defies the usage of dot-syntax in lua like here with `batman-adv`

> `<% if nodeinfo.software['batman-adv'] then -%>`

_Originally posted by @AiyionPrime in [#2214/.../status-page.html#L144](https://github.com/freifunk-gluon/gluon/pull/2214/files#diff-32907ffa1a5b6379d134e03669b9df3de2aab7d16fd543b22147208388c9a67eR144)

which is why I wouldn't change it lightheartedly.

Reply:
I agree it makes sense if it matches. There was a reason to not rename mesh-vpn to vpn, but I don't remember why...